src/math/CMakeLists.txt

```diff
@@ -14,6 +14,10 @@ if(CONFIG_MATH_DECIBELS)
         add_local_sources(sof decibels.c)
 endif()
 
+if(CONFIG_MATH_ARITHMETIC)
```

Comment:
You should use here the CONFIG_POWER_FIXED and CONFIG_BINARY_LOGARITHM_FIXED to separately add the source files conditionally to sof.